design/src/rfcs/rfc0041_identity_cache_partitions.md

```diff
@@ -0,0 +1,293 @@
+RFC: Identity Cache Partitions
+===============================
+
+> Status: RFC
+>
+> Applies to: AWS SDK for Rust
+
+Motivation
+-----------
+
+In the below example two clients are created from the same shared `SdkConfig` instance and each
+invoke a fictitious operation. Assume the operations use the same auth scheme relying on the same identity resolver.
+
+```rust,ignore
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+
+    let config = aws_config::defaults(BehaviorVersion::latest())
+        .load()
+        .await;
+    
+    let c1 = aws_sdk_foo::Client::new(&config);
+    c1.foo_operation().send().await;
+
+    let c2 = aws_sdk_bar::Client::new(&config);
+    c2.bar_operation().send().await;
+
+    Ok(())
+}
+```
+
+There are two problems with this currently.
+
+
+1. The identity resolvers (e.g. `credentials_provider` for SigV4) are re-used but we end up with a different 
+[`IdentityCachePartition`](https://github.com/smithy-lang/smithy-rs/blob/release-2024-03-25/rust-runtime/aws-smithy-runtime-api/src/client/identity.rs#L41)
+each time a client is created. 
+    * More specifically this happens every time a `SharedIdentityResolver` is [created](https://github.com/smithy-lang/smithy-rs/blob/release-2024-03-25/rust-runtime/aws-smithy-runtime-api/src/client/identity.rs#L197). The conversion from [`From<SdkConfig>`](https://github.com/awslabs/aws-sdk-rust/blob/release-2024-04-01/sdk/sts/src/config.rs#L1207) 
+    sets the credentials provider which [associates](https://github.com/awslabs/aws-sdk-rust/blob/release-2024-04-01/sdk/sts/src/config.rs#L960) it as
+    the identity resolver for the auth scheme. Internally this is [converted](https://github.com/awslabs/aws-sdk-rust/blob/release-2024-04-01/sdk/aws-smithy-runtime-api/src/client/runtime_components.rs#L663) to `SharedIdentityResolver` which creates the new partition. The end result is the credentials
```

Comment:
If it were already a `SharedIdentityResolver`, then the `IntoShared` implementation should detect that and wouldn't create a new instance. So that means it must be a `SharedCredentialsProvider` that's getting converted into a `SharedIdentityResolver`. This could be made clearer here.